src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1111,9 +1092,27 @@ SequenceDDLCommandsForTable(Oid relationId)
 						 "ALTER SEQUENCE %s OWNER TO %s", sequenceName,
 						 quote_identifier(ownerName));
 
+		/* commands to mark the object distributed in workers with metadata */
+		appendStringInfo(sequenceMarkDistStmt,
+						 "INSERT INTO citus.pg_dist_object (classid, objid, objsubid) "
+						 "(SELECT %u, c.oid, %d FROM pg_class c JOIN pg_namespace n ON (c.relnamespace = n.oid) "
+						 "WHERE c.relname = '%s' AND n.nspname = '%s') "
+						 "ON CONFLICT DO NOTHING;",
+						 RelationRelationId, 0,     /* objsubid is 0 */
+						 get_rel_name(sequenceOid), get_namespace_name(get_rel_namespace(
+																		   sequenceOid)));
+
+		sequenceDDLList = lappend(sequenceDDLList, sequenceUnMarkDistStmt->data);
 		sequenceDDLList = lappend(sequenceDDLList, wrappedSequenceDef->data);
 		sequenceDDLList = lappend(sequenceDDLList, sequenceGrantStmt->data);
+		sequenceDDLList = lappend(sequenceDDLList, sequenceMarkDistStmt->data);
+
+		/* get sequence address */
+		ObjectAddress sequenceAddress = { 0 };
+		ObjectAddressSet(sequenceAddress, RelationRelationId, sequenceOid);
+		EnsureDependenciesExistOnAllNodes(&sequenceAddress);
```

Comment:
Is there a way to do EnsureDependenciesExistOnAllNodes outside of SequenceDDLCommandsForTable? I would not expect this function to have side-effects.